### ISSUE TITLE
the training process is interrupted by “stopping early”

### ISSUE THREAD

**Issue by wangqipeng** [CSTR-Edinburgh/merlin#50]
Hi, all. 
     When i try to rebuild the US voice with slt_arctic, the process of training acoustic model is interrupted with the log of "stopping_early ".  It seems overfitting in the training process, so i set "dropout" to 0.5, it can 
save the acoustic model, but the voice synthesised by the merlin is just like "mur...mur...mur", nothing can be understand.
     i apply the default config in the s1/conf/general/acoustic_demo.conf. Any suggestion to improve the quality of synthesis will be appreciated!

**Comment by ronanki** [comment]
which demo examples you are using? 

For slt_arctic, the scripts for automatic preparation of config files and training are prepared in [egs/slt_arctic/s1]  (https://github.com/CSTR-Edinburgh/merlin/tree/master/egs/slt_arctic/s1). 

You can simply run the [full voice script] (https://github.com/CSTR-Edinburgh/merlin/blob/master/egs/slt_arctic/s1/run_full_voice.sh)

If in case, you see stopping_early, try to reduce learning rate first. If you want to use dropout, set it to a low value, may be around 0.1

**Comment by wangqipeng** [comment]
Thank you for your suggestion. Just now, I had downloaded the slt_arctic, run the demo script, the generated wav sounds well, the full data is running. Compared the configure file with full demo, i found the problem may be due to "train_file_number" which is set to 50 in the directory of "build_your_own_voice/s1/conf". So i reset it to 1000, and run again!

**Comment by dreamk73** [comment]
What confused me was that in the slt_arctic_demo it says 'stopping early' but it still wrote a model file to nnets_model. But when I tried my own db with 50 sentences, I didn't get a message that no model file was created. Maybe that is a good debug message to add?

**Comment by avashlin** [comment]
I also ran into this issue when trying to build a new voice, in the run_merlin.py file there's a check that only writes the best model found if it is after the 5th epoch. The stopping early criteria is activated when you validation loss increases after 15 epochs. In my case, the best model was found in the 4th epoch, and it didn't get better after that (I was only building on 40 files). I removed the 'only write the best model after the 5th epoch' check and my model file was written. The check is around line 325 in run_merlin.py
